pkg/scheduler/scheduler.go

```diff
@@ -0,0 +1,104 @@
+package scheduler
+
+import (
+	"sync"
+	"time"
+
+	"github.com/hashicorp/go-hclog"
+
+	"github.com/the-maldridge/nbuild/pkg/graph"
+	"github.com/the-maldridge/nbuild/pkg/types"
+)
+
+func NewScheduler(l hclog.Logger, c CapacityProvider, url string) *Scheduler {
+	x := Scheduler{
+		l:                l.Named("scheduler"),
+		capacityProvider: c,
+		apiClient:        graph.NewAPIClient(l),
+		queue:            make([]Build, 0),
+		queueMutex:       new(sync.Mutex),
+		tuples:           make([]types.SpecTuple, 0),
+	}
+	x.apiClient.Url = url
+
+	return &x
+}
+
+// Pops a build off the queue and hands it off to the CapacityProvider.
+func (s *Scheduler) send() bool {
```

Comment:
This function should probably return an Error rather than a bool, looking at how you are logging messages based on error return status.